modules/oe_authentication_user_fields/src/EventSubscriber/EuLoginAttributesToUserFieldsSubscriber.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_authentication_user_fields\EventSubscriber;
+
+use Drupal\cas\Event\CasPostLoginEvent;
+use Drupal\cas\Event\CasPreRegisterEvent;
+use Drupal\cas\Service\CasHelper;
+use Drupal\oe_authentication_user_fields\EuLoginAttributesHelper;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Copies the EU Login attributes to user fields.
+ */
+class EuLoginAttributesToUserFieldsSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      CasHelper::EVENT_POST_LOGIN => 'updateUserData',
+      CasHelper::EVENT_PRE_REGISTER => 'processUserProperties',
+    ];
+  }
+
+  /**
+   * Updates the user data based on the information taken from EU Login.
+   *
+   * @param \Drupal\cas\Event\CasPostLoginEvent $event
+   *   The triggered event.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   *   In case of failures an exception is thrown.
```

Comment:
The exception class seems to specific for a broad exception message.

Reply:
I am removing this entirely referring back to a discussion between Antonio and Pieter about not trying to document or catch exceptions related to the entity storage operations unless we are passing them as parameters when retrieving them.